04-onion-routing.md

```diff
@@ -292,7 +292,7 @@ The writer:
     - otherwise:
       - MUST set `total_msat` to the amount it wishes to pay.
     - MUST ensure that the total `amount_msat` of the HTLC set which arrives at the payee
-      is equal to `total_msat`.
+      is equal to or greater than `total_msat`.
```

Comment:
Now that we have #1032, shouldn't `amount_msat` be changed into the final `amt_to_forward`? To ensure that the sender remains in control of the set total?

Reply:
We could, that's a good point!

Reply:
I suppose it is still at the discretion of the receiver, but a premature release of the preimage may cause the receiver to receive less (although still meeting `total_msat`). With a premature release, routing nodes may short-cut pending htlcs.

Reply:
Yes that makes sense, don't hesitate to open a follow-up PR to fix that!

Reply:
Same discussion as we had here? https://github.com/lightning/bolts/pull/1032#discussion_r993310877

Reply:
Yes, but perhaps new is that we want to at least recommend the most incentive-compatible way in the spec?

Reply:
I agree, I opened #1040 to fix that.